pkg/querier/frontend/frontend.go

```diff
@@ -164,34 +164,46 @@ func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
 	resp, err := f.roundTripper.RoundTrip(r)
 	queryResponseTime := time.Since(startTime)
 
-	if f.cfg.LogQueriesLongerThan > 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
+	if err != nil {
+		writeError(w, err)
+	} else {
+		hs := w.Header()
+		for h, vs := range resp.Header {
+			hs[h] = vs
+		}
+		w.WriteHeader(resp.StatusCode)
+		io.Copy(w, resp.Body)
+	}
+
+	// If LogQueriesLongerThan is set to <0 we log every query
+	if f.cfg.LogQueriesLongerThan != 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
 		logMessage := []interface{}{
-			"msg", "slow query",
+			"msg", "slow query detected",
+			"method", r.Method,
 			"host", r.Host,
 			"path", r.URL.Path,
 			"time_taken", queryResponseTime.String(),
 		}
-		for k, v := range r.URL.Query() {
-			logMessage = append(logMessage, fmt.Sprintf("qs_%s", k), strings.Join(v, ","))
+
+		// Extract the ID from the header of the request
+		id, _, err := user.ExtractOrgIDFromHTTPRequest(r)
+		if err != nil {
+			logMessage = append(logMessage, "tenant_id", id)
 		}
-		pf := r.PostForm.Encode()
-		if pf != "" {
-			logMessage = append(logMessage, "body", pf)
+
+		// Ensure the form has been parsed so all the parameters are present
+		err = r.ParseForm()
+		if err != nil {
+			level.Warn(util.WithContext(r.Context(), f.log)).Log("unable to parse form for request")
 		}
-		level.Info(util.WithContext(r.Context(), f.log)).Log(logMessage...)
-	}
 
-	if err != nil {
-		writeError(w, err)
-		return
-	}
+		// Attempt to iterate through the Form to log any filled in values
+		for k, v := range r.Form {
+			logMessage = append(logMessage, fmt.Sprintf("qs_%s", k), strings.Join(v, ","))
```

Comment:
The field prefix is `qs_` which was originally done because we parsed only the query string. I'm wondering if we should rename it for clarity, ie. `param_`.

Reply:
I agree, I only kept `qs` since that is what was used previously.